### PR TITLE
Adds cooldown for robobuddies speak

### PR DESCRIPTION
**Pull request by Carbadox**
## About the PR 
Adds cooldown for robobuddies speak when using a gun on robobuddies. 

## Why's this needed? 
Without it, the buddies spew a paragraph of text, which means it extremely easy to spam the chat with it.

### CODE REVIEW

**Review by flrsh — COMMENTED** [review]
Could you add some user messages for when the cooldown is active? e.g. `You try to [src.locked ? "lock" : "unlock"] [src] but are too distracted. Try again in a sec.`

**Comment by Carbadox** [comment]
I realised that 5 seconds on the lock speak line might be too much, I'll lower it. Do you want me to still want me to add text when its in cooldown though?

**Comment by Yellow-Mushroom** [comment]
RIP getting screamed at with screens worth of speech getting thrown at you in 3 seconds

File: code/modules/robotics/bot/guardbot.dm [hunk 1]
```diff
@@ -826,16 +826,17 @@
...
-					src.locked = !src.locked
-					SPAWN_DBG(2 SECONDS)
-						speak(its_the_rd)
-						speak(long_day)
-						speak("Okay, everything's [src.locked ? "locked" : "unlocked"] now!")
-					return 1
-				else
-					return 0
+				if(!ON_COOLDOWN(src, "speak cooldown", 2 SECOND))
+					speak("Sorry, only people authorized by Thinktronic Data Systems may access my controls and accessories.")
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:830]: why is this not inside the Else below

**Comment by flrsh** [comment]
> I realised that 5 seconds on the lock speak line might be too much, I'll lower it. Do you want me to still want me to add text when its in cooldown though?

Yeah.

**Review by ZeWaka — CHANGES_REQUESTED** [review]
x

File: code/modules/robotics/bot/guardbot.dm (hunk 1 above)
Comment (pali6) [reply code/modules/robotics/bot/guardbot.dm:830]: Yeah the ifs here are not done correctly. Currently it'll show the distracted message and then lock/unlock anyway.

**Pull request closed without merging**